### PR TITLE
nx-table A11Y improvements - RSC-419

### PR DESCRIPTION
**Pull request by rpokorny**
https://issues.sonatype.org/browse/RSC-419

This PR wraps the contents of table sort headers and the clickable row chevron in a `<button>` element for better accessibility

### CODE REVIEW

File: gallery/src/components/NxTable/NxTableClickableExample.tsx [hunk 1]
```diff
@@ -32,6 +32,19 @@ const NxTableClickableExample = () => {
...
+            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed sed lorem vitae dolor mattis imperdiet a ac
+            nisl. Maecenas quis dapibus lacus, non lacinia dolor. In hac habitasse platea dictumst. Sed sit amet
+            nulla facilisis, elementum tellus quis, mattis turpis. Vivamus accumsan, purus vel maximus condimentum,
+            augue quam hendrerit velit, sed finibus nulla justo sit amet est. Nam tincidunt efficitur dapibus.
+            Quisque et pellentesque ante. Sed porttitor sem ipsum, sit amet blandit felis hendrerit nec. Suspendisse
+            congue, tortor tristique sollicitudin consectetur, sapien purus posuere massa, eget finibus nulla ipsum
+            sed sapien.
+          </NxTable.Cell>
+          <NxTable.Cell>Content</NxTable.Cell>
+          <NxTable.Cell chevron/>
```
Comment (glenhunter) [inline gallery/src/components/NxTable/NxTableClickableExample.tsx:46]: In Safari the Chevrons appear as buttons. 
Screen Shot 2021-04-28 at 10 58 18 AM

Comment (glenhunter) [reply gallery/src/components/NxTable/NxTableClickableExample.tsx:46]: Looks like you must override the UA styes with `border: none;` and `background-colour: transparent;`

File: gallery/src/styles/NxTable/NxTableSortableExample.tsx [hunk 2]
```diff
@@ -12,12 +12,14 @@ const NxTableSortable = () =>
...
     <thead>
       <tr className="nx-table-row nx-table-row--header">
-        <th className="nx-cell nx-cell--header nx-cell--sortable">
-          <span>Name</span>
-          <span className="nx-cell__sort-icons fa-layers">
-            <NxFontAwesomeIcon icon={faSortUp} />
-            <NxFontAwesomeIcon icon={faSortDown} />
-          </span>
+        <th className="nx-cell nx-cell--header nx-cell--sortable" aria-sort="none">
+          <button type="button" className="nx-cell__sort-btn" aria-label="Name unsorted">
```
Comment (glenhunter) [inline gallery/src/styles/NxTable/NxTableSortableExample.tsx:16]: Safari issues: 
Screen Shot 2021-04-28 at 11 15 30 AM

Solution is likely the same as the chevron button above.

Comment (glenhunter) [reply gallery/src/styles/NxTable/NxTableSortableExample.tsx:16]: It's worse, the sort icons are gone.

**Comment by rpokorny** [comment]
I'm not able to reproduce these Safari issues in browserstack.

**Comment by glenhunter** [comment]
> I'm not able to reproduce these Safari issues in browserstack.

I am no longer able to reproduce either issue.

File: gallery/src/styles/NxTable/NxTableSortableExample.tsx (hunk 2 above)
Comment (glenhunter) [reply gallery/src/styles/NxTable/NxTableSortableExample.tsx:16]: > It's worse, the sort icons are gone.

Checked on two computers, both running Safari 14.1 and I can lo longer reproduce.

File: gallery/src/components/NxTable/NxTableClickableExample.tsx (hunk 1 above)
Comment (glenhunter) [reply gallery/src/components/NxTable/NxTableClickableExample.tsx:46]: > In Safari the Chevrons appear as buttons.

Checked on two computers, both running Safari 14.1 and I can lo longer reproduce.

**Review by glenhunter — APPROVED** [review]

**Pull request merged**